### ISSUE TITLE
metager.org is not supported (Which search engines does it work with?)

### ISSUE THREAD

**Issue by DutchPete** [Endle/fireSeqSearch#127]
I installed the Tampermonkey script, quit and relaunched Logseq.

But I don't get any results from my Logseq graph listed. My search engine is Metger.

**Comment by Endle** [comment]
Do you mean https://metager.org/ ?

Currently this site is not in the list https://github.com/Endle/fireSeqSearch/blob/fa873eb722c4061447c90d7e614b5e30f683ece3/fireSeqSearch_addon/monkeyscript.user.js#L8

If you're interested, I can add the support to this site.

**Comment by DutchPete** [comment]
Yes, that is the search engine I mean.

I added it my self below line 8 as follows: `// @match        https://metager.org/?q=%s`.
That did not work, so I replaced `%s` by `*`. Still no result.
I then noticed, in line 132 (was 131 before) there is `debugStr: "tampermonkey",`: I replaced tampermonkey with violentmonkey, which is the extension I use.
Still no result.

I do not mean you to change the script. Just tell me which amendments I need to make, so if ever I want to change my default search engine I can make the changes in the script myself.

**Comment by Endle** [comment]
Thank you for your interest.


> I added it myself below line 8 as follows: `// @match https://metager.org/?q=%s`, which is what I have in my browser settings. That did not work, so I replaced `%s` by `*`. Still no result.
> 

The url structrue and DOM of metager is a bit different from Google. It seems that a bit more work is needed (just a bit)


> I then noticed, in line 132 (was 131 before) there is `debugStr: "tampermonkey",`: I replaced tampermonkey with violentmonkey, which is the extension I use. Still no result.
> 

I edited readme. Both tempermonkey and violentmonkey are supported.

> I do not mean you to change the script. Just tell me which amendments I need to make, so if ever I want to change my default search engine I can make the changes in the script myself.

Please eel free to submit feature requests, including adding support to more search engines! I'm more than happy to implement them.

I'm also welcome to PRs. My current code is still a little messy, which is unfriendly to new contributors. It would take time for refactoring, and more open issues by various people would increase my motivation :)

TLDR, I'll let you know when I finish this issue

**Comment by Endle** [comment]
Can you try the master branch https://github.com/Endle/fireSeqSearch/blob/master/fireSeqSearch_addon/violentmonkeyscript.user.js

**Comment by DutchPete** [comment]
It still does not work.

I don't know if it makes a difference that I use macOS and a Chromium browser.

**Comment by Endle** [comment]
macOS and Chromium are expected to work.

tampermonkey/violent monkey, which one you are using?

besides, have you upgraded to the newest server?

**Comment by Endle** [comment]
> What do you mean by “the newest server”?

Have you installed fireSeqSearch server?

I'm sorry that the readme file is a bit confusing. I updated it

**Comment by DutchPete** [comment]
Ah, OK, no I have not installed the server, I had not understood that. In any case, the ReadMe is clear now. Thank you.
